rust/experimental/query_engine/kql-parser/src/tabular_expressions.rs

```diff
@@ -1360,65 +1785,168 @@ mod tests {
+                                    QueryLocation::new_fake(),
+                                    "attributes",
+                                ),
+                            )),
+                            ScalarExpression::Static(StaticScalarExpression::Regex(
+                                RegexScalarExpression::new(
+                                    QueryLocation::new_fake(),
+                                    Regex::new("^namespace\\..*").unwrap(),
+                                ),
+                            )),
+                        ]),
+                    )],
+                )),
+            )],
+        );
+
+        // Note: In this test body is present in the ReduceMap because it
+        // matches the regex.
+        run_test_success(
+            "project-away b*",
+            vec![TransformExpression::ReduceMap(
+                ReduceMapTransformExpression::Remove(MapSelectionExpression::new_with_selectors(
                     QueryLocation::new_fake(),
                     MutableValueExpression::Source(SourceScalarExpression::new(
                         QueryLocation::new_fake(),
                         ValueAccessor::new(),
                     )),
                     vec![
-                        ScalarExpression::Static(StaticScalarExpression::Regex(
-                            RegexScalarExpression::new(
+                        MapSelector::KeyOrKeyPattern(ScalarExpression::Static(
+                            StaticScalarExpression::String(StringScalarExpression::new(
                                 QueryLocation::new_fake(),
-                                Regex::new(".*key.*value.*").unwrap(),
-                            ),
+                                "body",
+                            )),
                         )),
-                        ScalarExpression::Static(StaticScalarExpression::Regex(
-                            RegexScalarExpression::new(
+                        MapSelector::ValueAccessor(ValueAccessor::new_with_selectors(vec![
+                            ScalarExpression::Static(StaticScalarExpression::String(
+                                StringScalarExpression::new(
+                                    QueryLocation::new_fake(),
+                                    "attributes",
+                                ),
+                            )),
+                            ScalarExpression::Static(StaticScalarExpression::Regex(
+                                RegexScalarExpression::new(
+                                    QueryLocation::new_fake(),
+                                    Regex::new("^b.*").unwrap(),
+                                ),
+                            )),
+                        ])),
+                    ],
+                )),
+            )],
+        );
+
+        // In this test we understand attributes is removed by the regex so
+        // everything else gets dropped.
```

Comment:
This is an example of optimizations we can now perform having the schema information available 😄